### PR TITLE
Add ConnectionDraining to ApplicationGateway

### PR DESCRIPTION
**Pull request by jobatzil**
## Description

Added the support of ConnectionDraining for ApplicationGateways.

Along the AGW specific changes, the Api-Version of the Network part has been changed to 2016-12-01 and thus the test recordings were updated accordingly.

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

**Comment by markcowl** [comment]
@jobatzil You need to remove the app.config from your test project - this is why the build is failing

**Comment by jobatzil** [comment]
@markcowl  Okay, removed all app.config added by me.

File: src/ResourceManager/Network/Commands.Network/ApplicationGateway/ConnectionDraining/GetAzureApplicationGatewayConnectionDrainingCommand.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Network.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmApplicationGatewayConnectionDraining"),
+        OutputType(typeof(PSApplicationGatewayConnectionDraining))]
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/ApplicationGateway/ConnectionDraining/GetAzureApplicationGatewayConnectionDrainingCommand.cs]: returns an array of these

File: src/ResourceManager/Network/Commands.Network/packages.config [hunk 2]
```diff
@@ -19,8 +19,8 @@
   <package id="Microsoft.Data.Services.Client" version="5.6.4" targetFramework="net45" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.3" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime" version="2.3.4" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.4" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.5" targetFramework="net45" />
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/packages.config]: Unfortunately, this change needs to be made across the board, for all modules, changing in one module will just cause conflicts.  WIlll discuss how to deal with this change in the mornign before cutting the new branch

File: src/ResourceManager/Network/Commands.Network/ApplicationGateway/ConnectionDraining/GetAzureApplicationGatewayConnectionDrainingCommand.cs (hunk 1 above)
Comment (jobatzil) [inline src/ResourceManager/Network/Commands.Network/ApplicationGateway/ConnectionDraining/GetAzureApplicationGatewayConnectionDrainingCommand.cs]: It only returns one connection draining object.

Comment (jobatzil) [inline src/ResourceManager/Network/Commands.Network/ApplicationGateway/ConnectionDraining/GetAzureApplicationGatewayConnectionDrainingCommand.cs]: I remove the true I write object.

**Pull request merged**